.github/workflows/e2e.yml

```diff
@@ -0,0 +1,43 @@
+name: E2E Tests
+
+on: [push]
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: run hugo build
+        uses: jakejarvis/hugo-build-action@master 
+        with:
+          args: --gc --config ./config.toml -b http://localhost:1313
+      - name: run server
+        run: |
+          cd public && python -m http.server 1313 &
```

Comment:
But that means that these tests don't get run on each commit which sort of defeats the purpose of having tests doesn't it? Only you plan on deploying each branch someone makes to an environment to test and then tear it down.

Reply:
So, yes and no :upside_down_face:

My plan for deploy previews was a tiered approach.
1. Have them build on every pull request (not on JB infrastructure, but on netlify)
2. Have them run on merge into develop
3. Have them run on merge into prod

So, in that situation they would run on every push of commits, because netlify re-builds + GH actions re-run when commits are added to an already open PR.

(BTW, just to be on the same page I'm talking strictly about a future implementation)